src/gameplay/phases/common/votingMission.ts

```diff
@@ -270,6 +275,61 @@ class VotingMission implements IPhase {
   getProhibitedIndexesToPick(indexOfPlayer: number): number[] {
     return [];
   }
+
+  updateMissionSizesForSinad(): void {
+    // in 6p avalon, if m1 and m2 both succeed and m3 is a dani's pick (i.e. m3!=m2+1)
+    // then the sizes of m4 and m5 are swapped, requiring 4 ppl and 3 ppl respectively. 
+
+    if(!this.thisRoom.enableSinadMode)
+    {
+      return;
+    }
+    if(this.thisRoom.playersInGame.length !== 6 && this.thisRoom.gameMode !== GameMode.AVALON)
+    {
+      return; //not tested for other gamemodes. 
+    }
+
+    const VH = this.thisRoom.voteHistory; //for brevity
+
+    const setOfPlayersOnM2 = new Set<string>();
+    const setOfPlayersOnM3 = new Set<string>();
+
+
+    if( 
+      //m1 m2 pass, m3 failed, and we're at m4.1
+      this.thisRoom.missionHistory &&
+      this.thisRoom.missionHistory.length === 3 && 
+      this.thisRoom.missionHistory[0] === 'succeeded' &&
+      this.thisRoom.missionHistory[1] === 'succeeded' &&
+      this.thisRoom.missionHistory[2] === 'failed' &&
+      this.thisRoom.missionNum === 4 &&
+      this.thisRoom.pickNum === 1
```

Comment:
We should have a separate boolean tracking whether this has activated. This will help us avoid duplicate runs into this function. Although this code is only reached when a mission has voted, it'd be simpler and safer to just check for m1 success, m2 success, m3 fail, and then sinad hasn't activated yet to enter this function.

Reply:
Added a new method and field to track this. Hope [this](<https://github.com/vck3000/ProAvalon/pull/713/files#diff-fa564d9f0a3077d128a9edae956714e418036440f018e4c42849d44fca697ab4R2351-R2366>) works.